ert3/evaluator/_evaluator.py

```diff
@@ -39,7 +39,7 @@ def _prepare_input(
     transmitters: Dict[int, Dict[str, ert.data.RecordTransmitter]] = defaultdict(dict)
 
     futures = []
-    for input_ in step_config.input:
+    for input_ in step_config.input.values():
```

Comment:
Not sure here, but if `input_.record == record_name` then we can loop over step_config directly, ie:
```python
for record_name in step_config.input:
   for iens, record in enumerate(inputs.ensemble_records[record_name].records):
...
```

Reply:
Thanks. I think your solution is more human-readable and easier to intuitively understand, and will thus implement it, with the following changes: 
Assuming `input` is (kind of) a dict (that is immutable) with the field `record` (i.e. record_names) used as keys, I guess it would be 
```
for record_name in step_config.input.keys(): 
...
```

`input` is a special structure that inherits `Tuple`, and has a custom `values()` method that mimics that of a `dict`. I will implement `keys()` in the equivalent way.